core/shared/src/main/scala/coursier/util/Parse.scala

```diff
@@ -164,29 +171,38 @@ object Parse {
 
     val attrs = strings.drop(1).map({ x => {
       if (x.mkString.contains(argSeparator)) {
-        throw new ModuleParseError(s"'$argSeparator' is not allowed in attribute '$x' in '$s'. Please follow the format " +
+        return Left(s"'$argSeparator' is not allowed in attribute '$x' in '$s'. Please follow the format " +
           s"'org${argSeparator}name[${argSeparator}version][${argSeparator}config]${attrSeparator}attr1=val1${attrSeparator}attr2=val2'")
       }
       val y = x.split("=")
       if (y.length != 2) {
-        throw new ModuleParseError(s"Failed to parse attribute '$x' in '$s'. Keyword argument expected such as 'classifier=tests'")
+        return Left(s"Failed to parse attribute '$x' in '$s'. Keyword argument expected such as 'classifier=tests'")
```

Comment:
The `return`s could be avoided like the one below, although is less straightforward there. (Basically, we can `map` without using return, getting a sequence of eithers, then collect all the lefts on the one hand, and all the rights on the other, and check that the former is empty - but I can look into it later if you'd like, it's still better than throwing exceptions)

Reply:
Thank you so much for all of the tips!! It's been a while since I used scala and I forgot I could do all this stuff! I tried implementing what you suggested here but ran into some issues, so I thought I would fix the `isDefined`/`return Left(...)` issue and return to this when I have some more time if you haven't done it already. Does that sound alright to you?